boilerplate/app/components/Toggle.tsx

```diff
@@ -0,0 +1,493 @@
+      if (disabled) return colors.palette.neutral600
+      return colors.palette.neutral100
+    })()
+
+    const backgroundColor = withTiming(on ? onBackgroundColor : offBackgroundColor)
+    const start = withTiming(on ? "100%" : "0%")
+    const marginStart = withTiming(on ? -renderedKnobWidth : 0)
+
+    return { backgroundColor, start, marginStart }
+  }, [status, disabled, on])
+
+  const $switchContainerStyle = [
+    $switchContainer,
+    $animatedSwitchContainer,
+    $switchContainerStyleOverride,
+  ]
+  const $switchKnobStyle = [$switchKnob, $animatedSwitchKnob, $switchKnobStyleOverride]
+
+  return (
+    <Animated.View style={$switchContainerStyle}>
+      <SwitchAccessibilityLabel {...props} role="on" />
+      <SwitchAccessibilityLabel {...props} role="off" />
+
+      <Animated.View
+        style={$switchKnobStyle}
+        onLayout={(e) => setRenderedKnobWidth(Math.floor(e.nativeEvent.layout.width))}
+      />
+    </Animated.View>
+  )
+}
+
+function SwitchAccessibilityLabel(props: ToggleInputProps & { role: "on" | "off" }) {
+  const { on, disabled, status, switchAccessibilityMode, role, innerStyle, detailStyle } = props
+
+  if (!switchAccessibilityMode) return null
+
+  const shouldLabelBeVisible = (on && role === "on") || (!on && role === "off")
+
+  const $switchAccessibilityStyle = [
+    $switchAccessibility,
+    role === "off" && { end: "10%" },
+    role === "on" && { left: "10%" },
+  ]
+
+  const color = (function () {
+    if (disabled) return colors.palette.neutral600
+    if (status === "error" && !on) return colors.error
+    if (!on) return innerStyle?.backgroundColor || colors.palette.secondary500
+    return detailStyle?.backgroundColor || colors.palette.neutral100
+  })()
+
+  return (
+    <View style={$switchAccessibilityStyle}>
+      {switchAccessibilityMode === "text" && shouldLabelBeVisible && (
+        <Text
+          size="xs"
+          style={{ color }}
+          tx={role === "off" ? "accessibility.switchOff" : "accessibility.switchOn"}
+        />
+      )}
+
+      {switchAccessibilityMode === "icon" && shouldLabelBeVisible && (
+        <Image
+          style={[$switchAccessibilityIcon, { tintColor: color }]}
+          source={role === "off" ? iconRegistry.hidden : iconRegistry.view}
+        />
+      )}
+    </View>
+  )
+}
+
+function FieldLabel(props: ToggleProps) {
+  const {
+    label,
+    labelTx,
+    labelTxOptions,
+    LabelTextProps,
+    labelPosition,
+    labelStyle: $labelStyleOverride,
+  } = props
+
+  if (!label && !labelTx && !LabelTextProps?.children) return null
+
+  const $labelStyle = [
+    $label,
+    labelPosition === "right" && $labelRight,
+    labelPosition === "left" && $labelLeft,
+    $labelStyleOverride,
+    LabelTextProps?.style,
+  ]
+
+  return (
+    <Text
+      preset="formLabel"
+      text={label}
+      tx={labelTx}
+      txOptions={labelTxOptions}
+      {...LabelTextProps}
+      style={$labelStyle}
+    />
+  )
+}
```

Comment:
FIeldLabel is the component that renders text to the right or left of the Toggle input. It matches TextField's label style.